### ISSUE TITLE
Change the tag recognition to allow more than 1 tag to be used on a Vnet - allowing address ranges from two blocks to be reserved and consumed by the deployed vnet

### ISSUE THREAD

**Issue by jonprattLBG** [Azure/ipam#116]
**Is your feature request related to a problem? Please describe.**

- We have a number of Azure virtual networks, that consume IP ranges from 2 different Azure IPAM blocks. Some vnets even have 2 or more address space from 1 block and 1 or more from the other block.
- The problem we have - is 'because there only 1 X-IPAM-RES-ID tag - then only 1 reservation gets consumed' / network registered.
- The other reservations made for a virtual network dont get consumed.
- This leads to a mgmt overhead trying to track reservations that 'will never be consumed, but need to stay and never be removed' else an IP that is in use might be re-issued!

**Describe the solution you'd like**

- X-IPAM-RES-ID-1 & X-IPAM-RES-ID-2 to X-IPAM-RES-ID-N functionality, so many tags with a common start of name can be added to a virtual network, allowing multiple IPAM reservations to be consumed correctly.

**Describe alternatives you've considered**

- a manual tracking sheet that will not scale
- having 2 vnets (1 per block) and peering them together. At the scale we intend to get to, this will not be a good place to be.

**Comment by DCMattyG** [comment]
Hey there @jonprattLBG, thank you so much for reaching out, and for the excellent detail in your request.

First and foremost, you're not alone with this ask. We can definitely work on a way to help you accomplish this more seamlessly with Azure IPAM. For my better understanding, can you help us better understand some of the business reasons why you create Virtual Networks the way you described above (e.g. the function of two non-contiguous CIDR's assigned at creation)? This information will help us better align this tool to our customers business processes by understanding better they way each of you think about Azure Networking design/requirements.

I fully acknowledge your ask, but I was wondering if you've seen the discussion in #94, where there is another customer in a similar situation, but for now they are solving this challenge by creating the reservations, issuing a PUT on Block, then deleting the reservation. Is it ideal? Absolutely not, but I just want to find a way to unblock you for the time being, until we can work something up in the product that will meet your needs.

Hope all of this helps, I'll add the above to our project plan!

**Comment by DCMattyG** [comment]
Hey @jonprattLBG, I took a look at this closer and I think I found a very simple way to make this work.

The implementation would be adding N number of reservations to a Virtual Network TAG, separate by commas, like this:

![image](https://github.com/Azure/ipam/assets/12955732/d5b2c23c-b840-4107-9e55-14ed4a0a2f8a)

If that would work, I have this functioning in our DEV containers, and you're welcome to give it a whirl and provide some feedback. Here are the configuration changes you'd need to test the development version:

![image](https://github.com/Azure/ipam/assets/12955732/10664a3b-1f0d-458f-aa31-98ebd5880b0c)

Under the "Development Center" for the Azure App Service, add the "dev" suffix to the ACR address (e.g. *azureipamdev*) for all 3 containers. Click save, allow the App Service to pull the new containers and restart (usually ~5 minutes), then give it a try!

Excited for your feedback on this item, and thank you for your patience!

**Comment by gitfrs** [comment]
@DCMattyG 
Hello and thanks a lot for this update.
I took some time to test this. I'm using "XtratusCloud" Terraform Provider.
For me it's working great. Both reservation are now "seen" by IPAM.
So, once again, thanks you very much.

Just one little thing: In the UI, in Discover/VNETS, there is only one block displayed when there should be 2. But we won't use the UI, so for us, it's not an issue.

**Comment by DCMattyG** [comment]
@gitfrs, I can't tell you how much I appreciate your detailed testing. Displaying the multiple Blocks in the UI is something I definitely missed, so I'll get that addressed right away.

Thank you again and I'll keep you posted once these changes get pushed into Production.

**Comment by gitfrs** [comment]
@DCMattyG
Just to let you know, 
I went back to the non dev containers. I destroyed and redeployed my VNET and it still working properly.
To be able to test your dev containers, I have changed the way I create the Tags.

I guess I was doing it wrong before.

As I said, I'm using the terraform provider.
Here's what I'm doing : 

```hcl
locals {
  ipamtags = {X-IPAM-RES-ID:join(",", values(azureipam_reservation.cidr01.tags), values(azureipam_reservation.cidr02.tags))}
}
```

In the UI, everything seems ok. Can I check something else ?
Thanks

**Comment by gitfrs** [comment]
@DCMattyG Hello, 
Did you have any time looking into that ?
Is it normal that it worked in non-dev containers ?
Will this be included in the prod containers ?
I can do more testing if you need.
Thanks again.

**Comment by DCMattyG** [comment]
Hi @gitfrs, my immense apologies for not following-up on this. Everything is now incorporated into the current production containers. Please test everything out and let me know if you run into any issues.

Thanks for the reminder!

**Comment by gitfrs** [comment]
@DCMattyG Thanks for your answer. I'll get back on testing mid-august and let you know.
Thanks again :)

**Comment by DCMattyG** [comment]
Good morning @gitfrs, I wanted to see if you've had some time to thoroughly test this functionality and it's working per your expectations.

Let me know if there is anything that needs to be tweaked, otherwise let me know if I can close this issue out for now.

Really appreciate it!

**Comment by DCMattyG** [comment]
Closing this issue as this feature has been implemented in #134. Please don't hesitate to re-open this issue or create a new one should you need some additional support/clarification.

We're always here to help!
